FirebaseInstallations/Source/Library/InstallationsIDController/FIRInstallationsIDController.m

```diff
@@ -232,4 +224,56 @@ - (instancetype)initWithGoogleAppID:(NSString *)appID
       });
 }
 
+#pragma mark - Delete FID
+
+- (FBLPromise<NSNull *> *)deleteInstallation {
+  return [self.deleteInstallationPromiseCache getExistingPendingOrCreateNewPromise];
+}
+
+- (FBLPromise<NSNull *> *)createDeleteInstallationPromise {
+  // Check for ongoing requests first, if there is no a request, then check local storage for
+  // existing instlallation.
+  FBLPromise<FIRInstallationsItem *> *currentInstallationPromise =
+      [self.authTokenForcingRefreshPromiseCache getExistingPendingPromise]
+          ?: [self.authTokenPromiseCache getExistingPendingPromise]
+                 ?: [self.getInstallationPromiseCache getExistingPendingPromise]
+                        ?: [self getStoredInstallation];
```

Comment:
Not sure if it is fine from the readability perspective, but this is the most compact way to code "get the first no nil promise from the list".

Reply:
Should there be a function for extracting from the caches?  I don't have the context, but it seems strange to have three separate caches that have an implicit order.

Reply:
Those are not caches with data but a wrappers for operations to avoid performing the same async operation several times. So, basically, the code above means "check if the auth token is being refreshed then wait for it to complete and use the result, otherwise check for another operation. If there are no ongoing operations then just read from the storage".
Probably the naming is a bit confusing, so feel free to suggest a better one.